### PR TITLE
Add tox and mypy

### PR DESCRIPTION
**Pull request by mhchia**
### Done
- Add types in Pubsub partially 
- Add tox config file
- Put deps in `requirements_dev.txt` to `setup.py`
- Add mypy in deps
- Call linters pylint and mypy, and test in tox
- Change the structure of `.travis.yml`

### CODE REVIEW

**Comment by mhchia** [comment]
CI status
- [py37-test](https://travis-ci.com/libp2p/py-libp2p/jobs/219000593), all tests should pass and the codecov should be run in the end.
- [lint](https://travis-ci.com/libp2p/py-libp2p/jobs/219000594), `pylint` runs successfully and `mypy` fails as expected.

**Comment by mhchia** [comment]
This PR should be ready, cc @NIC619 .

File: setup.py [hunk 1]
```diff
@@ -26,6 +46,7 @@
         "lru-dict>=1.1.6",
         "aio_timers>=0.0.1,<0.1.0",
```
Comment (ChihChengLiang) [inline setup.py]: Looks like it no longer depends on "aio_timers"?

File: tox.ini [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    pytest --cov=./libp2p tests/
+    codecov
+basepython =
+    py37: python3.7
+
+[testenv:lint]
+basepython = python3
+extras = dev
+commands =
+    pylint --rcfile={toxinidir}/.pylintrc libp2p tests
```
Comment (ChihChengLiang) [inline tox.ini]: Should we lint `examples` too?

**Comment by mhchia** [comment]
@ChihChengLiang @NIC619 Thanks for the review. Will merge it if nothing goes wrong.

**Pull request merged**